### PR TITLE
[ISSUE #114] add orderly consume. resolve #114

### LINKED ISSUE #114: Add orderly consumer
**FEATURE REQUEST**

1. Please describe the feature you are requesting.
Add orderly consumer 

2. Provide any additional detail on your proposed use case for this feature.
Add orderly consumer 

2. Indicate the importance of this issue to you (blocker, must-have, should-have, nice-to-have). Are you currently using any workarounds to address this issue?
must-have

4. If there are some sub-tasks using -[] for each subtask and create a corresponding issue to map to the sub task:

### PR DESCRIPTION
**Pull request by xujianhai666**
## What is the purpose of the change

Add orderly consumer

## Brief changelog

Add orderly consumer

## Verifying this change

add example

### CODE REVIEW

File: consumer/consumer.go [hunk 1]
```diff
@@ -621,17 +634,19 @@ func (dc *defaultConsumer) updateProcessQueueTable(topic string, mqs []*primitiv
 
 	if dc.cType == _PushConsume {
 		for mq := range mqSet {
+			// BUG: the mq will send to channel, if not copy once, the next iter will modify the mq in the channel.
+			mq := mq
```
Comment (wenfengwang) [inline consumer/consumer.go]: rename left object

File: consumer/option.go [hunk 2]
```diff
@@ -100,12 +100,19 @@ type consumerOptions struct {
...
+	RebalanceLockInterval time.Duration
 }
 
 func defaultPushConsumerOptions() consumerOptions {
 	opts := consumerOptions{
-		ClientOptions: internal.DefaultClientOptions(),
-		Strategy:      AllocateByAveragely,
+		ClientOptions:              internal.DefaultClientOptions(),
+		Strategy:                   AllocateByAveragely,
+		MaxTimeConsumeContinuously: time.Duration(60000 * time.Millisecond),
```
Comment (wenfengwang) [inline consumer/option.go]: use `60 * time.Second`

File: consumer/option.go [hunk 3]
```diff
@@ -100,12 +100,19 @@ type consumerOptions struct {
...
+		RebalanceLockInterval:      20000 * time.Millisecond,
```
Comment (wenfengwang) [inline consumer/option.go]: 20 * time.Second

File: go.mod [hunk 4]
```diff
@@ -5,6 +5,7 @@ go 1.12
 require (
 	github.com/emirpasic/gods v1.12.0
 	github.com/golang/mock v1.3.1
+	github.com/luci/go-render v0.0.0-20160219211803-9a04cc21af0f
```
Comment (wenfengwang) [inline go.mod]: I can't see any code use it, could delete it?

File: consumer/consumer.go (hunk 1 above)
Comment (xujianhai666) [reply consumer/consumer.go]: done

File: consumer/option.go (hunk 3 above)
Comment (xujianhai666) [reply consumer/option.go]: done

File: go.mod (hunk 4 above)
Comment (xujianhai666) [reply go.mod]: done

File: consumer/option.go (hunk 2 above)
Comment (xujianhai666) [reply consumer/option.go]: done

**Comment by wenfengwang** [comment]
LGTM

**Pull request merged**